### ISSUE TITLE
"Asset must be part of at least one job" error in dagit when selecting a Dbt source

### ISSUE THREAD

**Issue by mingfang** [dagster-io/dagster#9520]
### Dagster version

dagit, version 1.0.6

### What's the issue?

"Asset must be part of at least one job" error in dagit when selecting a Dbt source.

### What did you expect to happen?

I should see details of the dbt sources.
Sources in dbt should behave like `SourceAssets` and definition they are not materialized by any jobs.

### How to reproduce?

To recreate the problem, change this line
https://github.com/dagster-io/dagster/blob/master/examples/assets_dbt_python/assets_dbt_python/repository.py#L51
to remove `raw_data_assets` like this 
`dbt_assets + forecasting_assets,`

Then in dagit `Global Asset Lineage` graph, try clicking `duckdb / raw_data / orders` and it results in this error
```
Operation name: SidebarAssetQuery

Message: Invariant failed. Description: Asset must be part of at least one job

Path: ["assetNodeOrError","type"]

Locations: [{"line":114,"column":3}]
```

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
@OwenKephart mind taking a look?

**Comment by mingfang** [comment]
@sryza @OwenKephart 
Thanks for looking into this issue.
I think this is a general issue with all `SourceAssets`.
You can see the problem by running this example
https://github.com/dagster-io/dagster/blob/master/docs/content/concepts/assets/software-defined-assets.mdx?plain=1#L126

**Comment by mingfang** [comment]
confirm fixed in v1.0.7
